### DISCUSSION TITLE
Enable sub properties of relationships to appear as columns

### DISCUSSION THREAD

**Discussion by samearcher** [jowilf/starlette-admin#307] [Q&A]
Hi,

I have a relationship, "mat_class" of my table, "A1_A3_Data", as follows:

```
class A1_A3_Data(A1_A3_Data_Base, ActiveRecordMixin, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    mat_class: Optional[Material_Class] = Relationship(back_populates="a1_a3_data")
```
This is in relationship (Many to one) with the Material_Class table:

```
class Material_Class(Material_Class_Base, ActiveRecordMixin, table=True):
    description: str = Field(min_length=3, max_length=200, unique=True)
    id: Optional[int] = Field(default=None, primary_key=True)
    a1_a3_data: List["A1_A3_Data"] = Relationship(back_populates="mat_class")
```

I'd like to be able to display (and sort) a column representing the mat_class description field as follows:

```
class A1_A3_Data_View(BaseModelView):
    fields = [
        A1_A3_Data.mat_class.description,
        "sheet_title",
        etc, etc
]
```

At the moment this gives the following error:

AttributeError: Neither 'InstrumentedAttribute' object nor 'Comparator' object associated with A1_A3_Data.mat_class has an attribute 'description'

I think this is because it's referencing the Class rather than the instances of the class?

Thanks

**Comment by Pauldic** [comment]
Facing the same issue here in 2024, it seems this is impossible with starlette-admin. I had Django background and still looking for any Admin plugin that can assist fill this gap
